src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/ui/workflow_yaml_editor.tsx

```diff
@@ -174,9 +174,19 @@ export const WorkflowYAMLEditor = ({
   // Refs / Disposables for Monaco providers
   const disposablesRef = useRef<monaco.IDisposable[]>([]);
   const focusedStepInfo = useSelector(selectFocusedStepInfo);
-  const yamlDocument = useSelector(selectYamlDocument);
   const workflowYamlSchemaLoose = useSelector(selectSchemaLoose);
-  const yamlDocumentRef = useRef<YAML.Document | null>(null);
+  // The current yaml document in the editor (could be unsaved)
+  const currentYamlDocument = useSelector(selectYamlDocument);
+
+  const yamlDocument = useMemo(() => {
```

Comment:
Can we set execution YAML to editor state so that we don't have much differences between execution yaml and being edited yaml?

Reply:
Yes! That's my plan, I'm thinking about moving the `execution` to the store so we centralize the branching logic (execution yaml vs editor yaml) in one place, and the computed values (document, lookup, definition) are generated with the correct one, so components will only need to trust the store. 
But that'll need some more work. This is just a quick patch for the weekly build.